vpr/src/route/route_tree.h

```diff
@@ -357,7 +358,16 @@ class RouteTree {
      * RouteTreeNode of the SINK it adds to the routing.
      * Locking operation: only one thread can update_from_heap() a RouteTree at a time. */
     std::tuple<vtr::optional<const RouteTreeNode&>, vtr::optional<const RouteTreeNode&>>
-    update_from_heap(t_heap* hptr, int target_net_pin_index, SpatialRouteTreeLookup* spatial_rt_lookup, bool is_flat);
+    update_from_heap(t_heap* hptr,
+                     int target_net_pin_index,
+                     SpatialRouteTreeLookup* spatial_rt_lookup,
+                     bool is_flat,
+                     const RouterLookahead& router_lookahead,
+                     t_conn_cost_params cost_params,
+                     const Netlist<>& net_list,
+                     const ParentNetId& net_id,
+                     int itry = -1,
+                     bool profile_lookahead = false);
```

Comment:
i am not a fan of default argument values

itry doesn't seem to need a default value because it is used only by the profiler.
I think we can get rid of `profile_lookahead`. The profiler is a global object, and it can store a flag to determine if it is enabled

Reply:
Where should I store this flag?